src/ML-PACE/compute_pace.h

```diff
@@ -0,0 +1,86 @@
+/* -*- c++ -*- ----------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#ifdef COMPUTE_CLASS
+// clang-format off
+ComputeStyle(pace,ComputePACE);
+// clang-format on
+#else
+
+#ifndef LMP_COMPUTE_PACE_H
+#define LMP_COMPUTE_PACE_H
+
+#include "compute.h"
+#include "ace-evaluator/ace_c_basis.h"
```

Comment:
These includes should not be here but rather used in compute_pace.cpp. It should be sufficient to have forward declaration of class instances in the header. Including application/library header files in a style class header carry a high potential for namespace pollution and unexpected compilation failures. Please see the corresponding notes in the programming style guide. https://docs.lammps.org/Modify_style.html#include-files-varied

If simple forward declarations cannot be used, the setup needs to be "PIMPL-ified", ie. a struct is used with all specific definitions and pointers and then the style header only needs to contain a pointer to that struct (with a forward declaration again) and the struct is then instantiated in the constructor. Please see pair style pacs for an example.

Reply:
Forward declarations are hard to use here so it has been "PIMPL-ified" instead in the most recent commit.

Reply:
We still need to update cmake build scripts and/or ACE library to fix the CI testing problem.  While we work on a more elegant solution for this, the non-standard ACE libraries can be used if the following D flags are specified during compilation:

-DPACELIB_URL= http://github.com/jmgoff/lammps-user-pace-1/archive/refs/tags/lammps-user-pace-compute.tar.gz
-DPACELIB_MD5=0616251507dffe83dbd38681d101aee8